pages/create-a-tool.tsx

```diff
@@ -0,0 +1,667 @@
+                    ) {
+                        fieldsFilled = false;
+                        needed = toolRequired[i][1] + ' Link"';
+                        setStillNeeded(needed);
+                        return fieldsFilled;
+                    }
+                }
+            }
+        }
+        //Checks the self check question fields
+        for (let k = 0; k < checkQuestion.length; k++) {
+            if (checkQuestion[k].question == "") {
+                //Checks the question title to ensure it isn't blank
+                fieldsFilled = false;
+                (needed =
+                    "Self Check Question " +
+                    checkQuestion[k].questionNumber.toString() +
+                    '\'s "Title"'),
+                    setStillNeeded(needed);
+                return fieldsFilled;
+            }
+            if (
+                checkQuestion[k].type == "multiple_choice" ||
+                checkQuestion[k].type == "multi_select"
+            ) {
+                //checks multiple choice/multiselect fields to ensure they aren't blank
+                for (let l = 0; l < checkQuestion[k].options.length; l++) {
+                    if (checkQuestion[k].options[l][0] == "") {
+                        fieldsFilled = false;
+                        (needed =
+                            "Self Check Question " +
+                            checkQuestion[k].questionNumber.toString() +
+                            '\'s "User Facing Options"'),
+                            setStillNeeded(needed);
+                        return fieldsFilled;
+                    }
+                    if (checkQuestion[k].options[l][1] == "") {
+                        fieldsFilled = false;
+                        (needed =
+                            "Self Check Question " +
+                            checkQuestion[k].questionNumber.toString() +
+                            '\'s "Corresponding Values"'),
+                            setStillNeeded(needed);
+                        return fieldsFilled;
+                    }
+                }
+            } else {
+                needed = "";
+            }
+        }
+        setStillNeeded(needed);
+        return fieldsFilled;
+    };
+
+    //self check card
+    const defaultQuestions = [
+        {
+            _id: "60e642d7e4a1ae34207a92a0",
+            type: "multiple_choice",
+            question: "",
+            options: [
+                ["", ""],
+                ["", ""],
+                ["", ""],
+            ],
+            alphanumericInput: true,
+            questionNumber: 1,
+        },
+    ];
+    const [questionList, setQuestionList] = useState(defaultQuestions);
+    const [lastSavedQuestions, setLastSavedQuestions] = useState(
+        JSON.parse(JSON.stringify(defaultQuestions)),
+    );
+    //keeps track of count of questions made so each question has a unique id
+    const [count, setCount] = useState(2);
```

Comment:
Quick question why do we use a starting state of 2 for question count?

Reply:
Because that controls the given question ID and the first question ID starts at 1 I believe

Reply:
Okay that makes sense- by the questionID are you referring to what's on line 186 as the `defaultQuestions` id?